src/background/sidebar-injector.js

```diff
@@ -39,19 +39,17 @@ function addJSONScriptTag(name, content) {
  * chrome.tabs.executeScript() into the main frame of a page.
  *
  * executeScript() returns an array of results, one per frame which the script
- * was injected into.
+ * was injected into. If the frame fails to execute the script it returns a value
+ * of `null`.
  *
  * See https://developer.chrome.com/extensions/tabs#method-executeScript
  *
- * @param {Array<any>?} result
+ * @template T
+ * @param {Array<T|null>} result
  */
 function extractContentScriptResult(result) {
-  if (Array.isArray(result) && result.length > 0) {
+  if (result.length > 0) {
     return result[0];
-  } else if (typeof result === 'object') {
-    // Firefox currently returns an object instead of
-    // an array from executeScript()
-    return result;
   } else {
     return null;
```

Comment:
I'm not sure if this branch can ever actually be hit in practice, because every tab presumably has at least one frame.